flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPoolTest.java

```diff
@@ -637,13 +672,17 @@ public void go() throws Exception {
                                             numberOfSegmentsToRequest));
                         }
                     };
-            asyncRequest.start();
 
             // recycle 5 segments
             CompletableFuture<?> availableFuture = globalPool.getAvailableFuture();
             globalPool.recycleUnpooledMemorySegments(segments1);
             assertTrue(availableFuture.isDone());
 
+            // Starting to requesting next segments only after previous segments were release
+            // otherwise the race condition between requesting new segments and releasing old ones
+            // are possible:
+            asyncRequest.start();
```

Comment:
Does the async request actually make sense here? I'm not sure if the async request adds something relevant to the test case?

Reply:
In addition, isn't the use of ArrayList in a concurrent thread non-thread-safe? (CopyOnWriteArrayList would be a thread-safe alternative.)

Reply:
Actually, right now I see that this test was incorrect. This test supposed that `requestUnpooledMemorySegments` would be blocked until enough buffers will be available but it is just not true. I don't know it was the initial mistake in the test or the old behavior was different. Anyway I will remove this extra thread that doesn't make sense anymore.